src/SIM800-AT-Zephyr.cpp

```diff
@@ -51,19 +51,26 @@ int GPRS::ip_rx_data(void)
     while(tcp_packet_len != tcp_avail) {
         tcp_packet_len = tcp_avail;
         send_cmd(cmd, DEFAULT_TIMEOUT, "OK");
-        sscanf(resp_buf, " +CIPRXGET: 4,%hu,0", &tcp_avail);
+        // ACK string position varies between 2 and three characters
+        // with mixed whitespace so we can't sscanf for it properly.
+        char *ack_pos = strstr(resp_buf, "+CIPRXGET:");
+        if (NULL != ack_pos) {
+            sscanf(ack_pos, "+CIPRXGET: 4,%hu", &tcp_avail);
+        }
         k_sleep(K_MSEC(150));
     }
 
     // The length of output data can not exceed 1460 bytes at a time,
     // based on SIM800 Series_AT Command Manual.
     // We assume the worst case transfer speed of 9.6kbps, equivalent
     // to ~1byte/ms. We also append a margin of 55ms.
+    if(tcp_packet_len > 1460) tcp_packet_len = 1460;
```

Comment:
Please use curlies for one liners, eg:

```suggestion
if(tcp_packet_len > 1460) {
    tcp_packet_len = 1460;
}
```